### PR TITLE
build: support having more than one ICU in use

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

This allows us to cross-compile the standard library to foreign targets on a
single host.  The ICU dependencies can be specified on the command line on a
per-target basis.  If one is not specified, we fall back to the default search
path and use that for the other targets.

Special thanks to Dimitri Gribenko for the various hints in getting this wired
up.

### CODE REVIEW

**Comment by compnerd** [comment]
CC @gribozavr @modocache

**Comment by gribozavr** [comment]
Thank you!  I haven't looked at the patch itself yet, but:

@swift-ci Please test

**Comment by modocache** [comment]
This is an awesome change, thank you! Android build also continues to work with these changes.

File: stdlib/public/stubs/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +22,7 @@ add_swift_library(swiftStdlibStubs OBJECT_LIBRARY TARGET_LIBRARY
   UnicodeExtendedGraphemeClusters.cpp.gyb
   ${swift_stubs_objc_sources}
   ${swift_stubs_unicode_normalization_sources}
+  TARGET_SDKS ${SWIFT_SDKS}
```
Comment (compnerd) [inline stdlib/public/stubs/CMakeLists.txt]: Pretty sure that this hunk is bogus.  It is already a `TARGET_LIBRARY`, so I dont see why this needs `TARGET_SDKS` as well.  Thoughts @gribozavr?

**Comment by compnerd** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
ping @gribozavr ... the failure seems unrelated, its in the SDK overlay for HomeKit on watchOS.

**Comment by gribozavr** [comment]
@swift-ci Please test OS X platform

**Comment by gribozavr** [comment]
Thank you!  LGTM as long as CI passes.

**Pull request merged**